app/models/exam_version.rb

```diff
@@ -39,12 +39,14 @@ def answers
   end
 
   def default_answers
+    def_answers = answers.map do |ans_q|
+      ans_q.map do |ans_p|
+        ans_p.map { |_| { "NO_ANS": true } }
+      end
+    end
+    def_answers = [[[]]] if def_answers.empty?
```

Comment:
How could this last line be needed?  And how is it of any help, if it implies there's exactly one question with exactly one part with exactly one unanswered body-item?

Reply:
It fixes showing a question/part without sub-parts/items - we rely on answers being a 3d array in multiple places, and indexing with `[]` on undefined throws an error.